### PR TITLE
Tip: Fat arrow functions

### PR DESCRIPTION
**Pull request by pklinger**
Please review and give some tips for improvements, thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,95 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter
...
+### Simple syntax example
+Have a look at these two code snippets, which exactly do the same job. You will quickly understand what fat arrow functions do.
+
+```javascript
+// general syntax for fat arrow functions
+(param) => expression
+
+
+// using functions
+var arr = [5,3,2,9,1];
```
Comment (parro-it) [inline README.md:39]: Don't mix let & var

File: README.md [hunk 2]
```diff
@@ -18,6 +18,95 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter
...
+let arrFunc = arr.map(function(x) {
+  return x * x;
+});
+console.log(arr)
+
+// using fat arrow
+var arr = [5,3,2,9,1];
+let arrFunc = arr.map((x) => x*x);
```
Comment (parro-it) [inline README.md:47]: You can omit () for a single argument

File: README.md [hunk 3]
```diff
@@ -18,6 +18,95 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter
...
+console.log(arr)
+```
+
+As you may see, the fat arrow function in this case may save you time typing out the parentheses as well as the function and return keywords. I would advice you to always write parentheses around the parameter inputs as the parentheses will be needed for multiple input parameters such as in `(x,y) => x+y` anyways. It is just a way to cope with forgetting them in different use cases. But the code above would also work like this: `x => x*x`. So far these are only syntactical improvements, which lead to less LOC and better readability.
```
Comment (parro-it) [inline README.md:51]: Uhm, I would prefer to omit the parentheses for a single arg

File: README.md [hunk 4]
```diff
@@ -18,6 +18,95 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter
...
+As you may see, the fat arrow function in this case may save you time typing out the parentheses as well as the function and return keywords. I would advice you to always write parentheses around the parameter inputs as the parentheses will be needed for multiple input parameters such as in `(x,y) => x+y` anyways. It is just a way to cope with forgetting them in different use cases. But the code above would also work like this: `x => x*x`. So far these are only syntactical improvements, which lead to less LOC and better readability. 
+
+### Lexically binding `this`
+
+There is another good reason to use fat arrow functions. There is the issue with the context of `this`. With arrow functions, you will not worry about `.bind(this)` or setting `that = this` anymore, as fat arrow functions pick the context of `this` from the lexical sourrounding. Have a look at the next [example] (https://jsfiddle.net/pklinger/rw94oc11/):
+
+```javascript
+
+// globally defined this.i
+this.i = 100;
```
Comment (parro-it) [inline README.md:60]: I think this is clearer if you write it as `global.i`

File: README.md (hunk 3 above)
Comment (pklinger) [inline README.md:51]: Thats perfectly valid, too. I pointed out, why I would still put them there. I guess its a personal style thing?

Comment (parro-it) [inline README.md:51]: Yes, it is. I wrote comments while reading, so see the note after I wrote first comment regarding parentheses :smile_cat:

**Pull request closed without merging**